### PR TITLE
Handle custom string not equal case for Pinot

### PR DESCRIPTION
**Pull request by bowenxia**
**What changed?**
- Handle custom string not equal case for Pinot in pinot_query_validator
- Deleted redundant parenthesis for and clause
- Updated test cases accordingly

**Why?**
- It is a bug. Previously we didn't catch this edge case. 
For example, when search for --query "CustomStringField != 'xxx'", it had the same result as "CustomStringField = 'xxx'"
- To make it the same behavior as ES

**How did you test it?**
Unit test, manually test in QueryBuilder

**Potential risks**
no

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/pinot/pinotQueryValidator.go [hunk 1]
```diff
@@ -453,21 +453,22 @@ func processCustomKeyword(operator string, colNameStr string, colValStr string)
 		colNameStr, operator, colValStr, colNameStr, operator, colValStr)
 }
 
-func processCustomString(comparisonExpr *sqlparser.ComparisonExpr, colNameStr string, colValStr string) string {
-	// change to like statement for partial match
-	comparisonExpr.Operator = sqlparser.LikeStr
```
Comment (bowenxia) [inline common/pinot/pinotQueryValidator.go:458]: This part was a legacy code from the first version of pinot_query_validator. Forgot to delete previously.

File: common/pinot/pinotQueryValidator_test.go [hunk 2]
```diff
@@ -301,6 +301,14 @@ func TestValidateQuery(t *testing.T) {
 			validated: "",
 			err:       "failed to process a bool key to SQLVal: &{<nil> abc { }}",
 		},
+		"case22-1: test not equal to a string field": {
```
Comment (shijiesheng) [inline common/pinot/pinotQueryValidator_test.go:304]: Could you add cases
* CustomStringField is missing
* CustomStringFieldA != 'abc' AND CustomStringFieldB != 'def'
* CustomStringFieldA = 'abc' AND CustomStringFieldB != 'def'
* CustomStringFieldA != 'abc' AND CustomStringFieldB = 'def'
* CustomStringFieldA = 'abc' AND CustomStringFieldB = 'def'
* CustomStringFieldA != 'abc' OR CustomStringFieldB != 'def'
* CustomStringFieldA = 'abc' OR CustomStringFieldB != 'def'
* CustomStringFieldA != 'abc' OR CustomStringFieldB = 'def'
* CustomStringFieldA = 'abc' OR CustomStringFieldB = 'def'

**Comment by davidporter-id-au** [comment]
@bowenxia  it's not super clear to me what the bug actually was (sorry, I may have not had enough context or missed something). Do you mind adding an example of a query that was failing ?

**Review by shijiesheng — APPROVED** [review]
